crates/viewer/re_viewer/src/app.rs

```diff
@@ -779,6 +772,176 @@ impl App {
+        let store_sources = store_hub
+            .store_bundle()
+            .entity_dbs()
+            .filter_map(|db| db.data_source.as_ref());
+        let mut all_sources = store_sources.chain(active_sources.iter().map(|s| s.as_ref()));
+
+        match data_source {
+            DataSource::RrdHttpUrl { url, follow } => {
+                let new_source = SmartChannelSource::RrdHttpStream {
+                    url: url.clone(),
+                    follow: *follow,
+                };
+                if all_sources.any(|source| source.is_same_ignoring_uri_fragments(&new_source)) {
+                    if let Some(entity_db) = store_hub.find_recording_store_by_source(&new_source) {
+                        if *follow {
+                            let rec_cfg = self.state.recording_config_mut(entity_db);
+                            let time_ctrl = rec_cfg.time_ctrl.get_mut();
+                            time_ctrl.set_play_state(
+                                entity_db.times_per_timeline(),
+                                PlayState::Following,
+                            );
+                        }
+
+                        let store_id = entity_db.store_id().clone();
+                        drop(all_sources);
+                        self.make_store_active_and_highlight(store_hub, egui_ctx, &store_id);
+                    }
+                    return;
+                }
+            }
+
+            #[cfg(not(target_arch = "wasm32"))]
+            DataSource::FilePath(_file_source, path) => {
+                let new_source = SmartChannelSource::File(path.clone());
+                if all_sources.any(|source| source.is_same_ignoring_uri_fragments(&new_source)) {
+                    if let Some(entity_db) = store_hub.find_recording_store_by_source(&new_source) {
+                        let store_id = entity_db.store_id().clone();
+                        if store_id.is_recording() {
+                            drop(all_sources);
+                            self.make_store_active_and_highlight(store_hub, egui_ctx, &store_id);
+                        }
+                    }
+                    return;
+                }
+            }
+
+            DataSource::FileContents(_file_source, _file_contents) => {
+                // For raw file contents we currently can't determine whether we're already receiving them.
+            }
+
+            #[cfg(not(target_arch = "wasm32"))]
+            DataSource::Stdin => {
+                let new_source = SmartChannelSource::Stdin;
+                if all_sources.any(|source| source.is_same_ignoring_uri_fragments(&new_source)) {
+                    if let Some(entity_db) = store_hub.find_recording_store_by_source(&new_source) {
+                        let store_id = entity_db.store_id().clone();
+                        drop(all_sources);
+                        self.make_store_active_and_highlight(store_hub, egui_ctx, &store_id);
```

Comment:
same? (maybe this bit of code could go to a local fn or closure to avoid duplication?)